src/if_statements.py

```diff
@@ -182,167 +268,74 @@ def label_for_node(context: Context, node: Node) -> str:
         return f"block_{node.block.index}"
 
 
-def emit_node(context: Context, node: Node, body: Body, indent: int) -> None:
-    """Emit a node, together with a label for it (which is only printed if
-    something jumps to it, e.g. currently for loops)."""
-    if isinstance(node, ReturnNode) and not node.is_real():
-        body.add_node(node, indent, comment_empty=False)
+def emit_node(context: Context, node: Node, body: Body) -> bool:
+    """
+    Try to emit a node for the first time, together with a label for it.
+    The label is only printed if something jumps to it, e.g. a loop.
+
+    For return nodes, it's preferred to emit multiple copies, rather than
+    goto'ing a single return statement.
+
+    For other nodes that were already emitted, instead emit a goto.
+    Since nodes represent positions in assembly, and we use phi's for preserved
+    variable contents, this will end up semantically equivalent. This can happen
+    sometimes when early returns/continues/|| are not detected correctly, and
+    this hints at that situation better than if we just blindly duplicate the block
+    """
+    if node in context.emitted_nodes:
+        # TODO: Treating ReturnNode as a special case and emitting it repeatedly
+        # hides the fact that we failed to fold the control flow. Maybe remove?
```

Comment:
Certainly we should emit a comment at the very least indicating duplicate emission?